conans/model/scm.py

```diff
@@ -1,34 +1,51 @@
-import json
 import os
 import subprocess
 
+from six import string_types
+
 from conans.client.tools.scm import Git, SVN
 from conans.errors import ConanException
 from conans.util.files import rmdir
 
 
 def get_scm_data(conanfile):
-    try:
+    data = getattr(conanfile, "scm", None)
+    if data is not None and isinstance(data, dict):
         return SCMData(conanfile)
-    except ConanException:
+    else:
         return None
 
 
+def _get_dict_value(data, key, expected_type, default=None, disallowed_type=None):
+    if key in data:
+        r = data.get(key)
+        if r is None:  # None is always a valid value
+            return r
+        if not isinstance(r, expected_type) or (disallowed_type and isinstance(r, disallowed_type)):
+            type_str = "' or '".join([it.__name__ for it in expected_type]) \
+                if isinstance(expected_type, tuple) else expected_type.__name__
+            raise ConanException("SCM value for '{}' must be of type '{}'"
+                                 " (found '{}')".format(key, type_str, type(r).__name__))
+        return r
+    return default
+
+
 class SCMData(object):
+    VERIFY_SSL_DEFAULT = True
+    SHALLOW_DEFAULT = True
 
     def __init__(self, conanfile):
-        data = getattr(conanfile, "scm", None)
-        if data is not None and isinstance(data, dict):
-            self.type = data.get("type")
-            self.url = data.get("url")
-            self.revision = data.get("revision")
-            self.verify_ssl = data.get("verify_ssl")
-            self.username = data.get("username")
-            self.password = data.get("password")
-            self.subfolder = data.get("subfolder")
-            self.submodule = data.get("submodule")
-        else:
-            raise ConanException("Not SCM enabled in conanfile")
+        data = getattr(conanfile, "scm")
+        self.type = _get_dict_value(data, "type", string_types)
+        self.url = _get_dict_value(data, "url", string_types)
+        self.revision = _get_dict_value(data, "revision", string_types + (int, ),
+                                        disallowed_type=bool)  # bool is subclass of integer
```

Comment:
I leave `int` as a valid type for revision because SVN users might be using it this way (but I don't want a `bool` to be valid input).